x-pack/plugins/timelines/public/components/t_grid/standalone/index.tsx

```diff
@@ -107,6 +107,7 @@ export interface TGridStandaloneProps {
   itemsPerPageOptions: number[];
   query: Query;
   onRuleChange?: () => void;
+  onTGridStateChange?: (state: State) => void;
```

Comment:
Nit: what about simply `onStateChange`? None of the other props references the `TGrid` explicitly. This one somewhat couples the component interface (props) with the concrete implementation.